scripts/resolve_container_path.py

```diff
@@ -19,24 +29,46 @@
 
 def resolve_container_path(container_path):
     container_path = os.path.realpath(container_path)
+    mountinfo_path = '/proc/self/mountinfo'
     try:
-        with open('/proc/self/cgroup') as f:
-            # Entries in /proc/self/cgroup look like this (note the nesting):
-            # 11:name=systemd:/docker/82c1bd2…23b5bcf/docker/6547bce…60ca5a7
-            prefix, container_id = next(f).strip().split(':')[2].split('/')[-2:]
+        with open(mountinfo_path) as f:
+            log.info('Found %s', mountinfo_path)
+            # Entries of interest in /proc/self/mountinfo look as follows:
+            # 752 744 259:2 /docker/containers/dc61d9…CONTAINER_ID…/<variable> …
+            expected_prefix = '/docker/containers'
+            log.info('cgroup v2 prefix is %s', expected_prefix)
+            container_ids = set()
+            for line in f.readlines():
+                if expected_prefix in line:
+                    parts = line.split(' ')
```

Comment:
Each line ends in a newline character which you should remove prior to splitting the line. Otherwise, the last field will have newline character. You don't care about the last field here, but nevertheless should code the parsing correctly so that the code could be extended without surprises, especially since the removing of a single character is cheap. Note, that it would be an error for a line to not end in newline.